### PR TITLE
support vim augroups

### PR DESCRIPTION
**Pull request by catlee**
This adds `augroup END` after beginning a new `augroup`. It ignores `augroup end` or `augroup END`.

Unfortunately, I can't get the tests working locally.

### CODE REVIEW

**Comment by RRethy** [comment]
Closing this PR and pasting the rational that was shared over slack about why `augroup` won't work. If any future reader thinks of a way to get augroup to work (without using regex which I'm not going to add to this plugin), feel free to comment the solution and I can take another look.

I took a closer look at the syntax tree for vimscript and I don’t think augroup is possible. If we take a look at this test case, it’ll fail since every time you press <CR> after augroup foo, it’ll produce a augroup END.

```ruby
test "vim, augroup", <<~END
-augroup foo█
-augroup END
+augroup foo
+  
+augroup END
END
```

The primary challenge is that augroup <ident> is a valid statement, as are all of the following:

```
augroup <ident>
augroup <ident>
augroup <ident>
augroup END
augroup END
augroup END
```

nvim-treesitter-endwise works under the assumption that the syntax tree is in an invalid state, it’ll then only add an end node if it helps put the parser into a more valid state. However, since vimscript `augroup <ident>` is always valid, even if `<ident>` is END, we don’t have a way of knowing if we should be adding augroup END. The best way I could illustrate this to myself is with the following vimscript:

```vim
aug foo
aug bar
autocmd VimLeave * echo 'bar'
augroup END
autocmd VimLeave * echo 'foo'
```


It’ll produce the following syntax tree (viewed using :TSPlayground)

```
augroup_statement [0, 0] - [0, 7]
  augroup_name [0, 4] - [0, 7]
augroup_statement [1, 0] - [1, 7]
  augroup_name [1, 4] - [1, 7]
autocmd_statement [2, 0] - [2, 29]
  au_event_list [2, 8] - [2, 16]
    au_event [2, 8] - [2, 16]
  pattern [2, 17] - [2, 18]
  command: echo_statement [2, 19] - [2, 29]
    string_literal [2, 24] - [2, 29]
augroup_statement [3, 0] - [3, 11]
  augroup_name [3, 8] - [3, 11]
autocmd_statement [4, 0] - [4, 29]
  au_event_list [4, 8] - [4, 16]
    au_event [4, 8] - [4, 16]
  pattern [4, 17] - [4, 18]
  command: echo_statement [4, 19] - [4, 29]
    string_literal [4, 24] - [4, 29]
```

If you then execute that valid vimscript, the autocmd `autocmd VimLeave * echo 'foo'` will be under the anon augroup. This is because wil augroup foo is never ended, it’s replaced by augroup bar since augroups are not hierarchical. While vimscript is executed, the augroup is effectively a global variable, this can also be seen by sourcing another file which just has the line augroup baz and seeing what augroup owns the autocommands which you create (it’ll be the baz augroup even tho this was from another file). This baz augroup can then be closed with augroup END which makes this valid and resonable.

```vim
source ./baz.vim
augroup END
```

If baz.vim just contains augroup baz.

TLDR `augroup <ident>` is a statement that is valid on it’s own and thus doesn’t work with nvim-treesitter-endwise. Even if we just operated like vim-endwise and regex’d everything, it still would have flaws (the last example above) since the current augroup is effectively a global variable in the vimscript parser (built into vim/neovim).
